src/iface/route.rs

```diff
@@ -74,16 +74,36 @@ impl<'a> Routes<'a> {
         Routes { storage }
     }
 
-    /// Update the routes of this node.
-    pub fn update<F: FnOnce(&mut ManagedMap<'a, IpCidr, Route>)>(&mut self, f: F) {
-        f(&mut self.storage);
+    /// Add a new route.
+    ///
+    /// On success, return the previous route, if any.
+    pub fn add_route(&mut self, cidr: IpCidr, route: Route) -> Result<Option<Route>> {
```

Comment:
The naming of this function is inconsistent with `set_*_default_route`. Either this should be called `set_route`, or those functions should be called `add_*_default_route`. They fundamentally do the same thing.